bindings/java/src/org/sleuthkit/datamodel/Blackboard.java

```diff
@@ -123,7 +146,93 @@ public void postArtifacts(Collection<BlackboardArtifact> artifacts, String modul
 			throw new BlackboardException("Failed to get or add artifact type", ex);
 		}
 	}
-
+	
+	/**
+	 * Adds new analysis result artifact.
+	 *
+	 * @param artifactType      Type of analysis result artifact to create.
+	 * @param obj_id            Object id of parent.
+	 * @param datasource_obj_id Data source object id.
+	 * @param score	            Score associated with this analysis result.
+	 * @param conclusion        Conclusion of the analysis, may be null or an
+	 *                          empty string.
+	 * @param configuration     Configuration association with this analysis,
+	 *                          may be null or an empty string.
+	 * @param justification     Justification, may be null or an empty string.
+	 * @param attributesList    Attributes to be attached to this analysis
+	 *                          result artifact.
+	 * @param transaction       DB transaction to use.
+	 *
+	 * @return Analysis result created.
+	 *
+	 * @throws BlackboardException exception thrown if a critical error occurs
+	 *                             within TSK core
+	 */
+	public AnalysisResult newAnalysisResult(BlackboardArtifact.Type artifactType, long obj_id, long datasource_obj_id, Score score, String conclusion, String configuration, String justification, Collection<BlackboardAttribute> attributesList, CaseDbTransaction transaction) throws BlackboardException {
+		
+		try {
+			AnalysisResult analysisResult =  caseDb.newAnalysisResult(artifactType, obj_id, datasource_obj_id, score, conclusion, configuration, justification, transaction.getConnection());
+			
+			// add the given attributes
+			addBlackboardAttributes(attributesList, artifactType.getTypeID(), analysisResult.getArtifactID(), transaction);
```

Comment:
@raman-bt  I think this call is not a very good idea. I added this function but it is not appropriate - I think I will need to change it (thinking about the how to ) 
The issue here is that the attributes list will in fact get persisted, but the AnalysisResutl (the underlying BlackboardArtifact) will not cache the attributes that we just added. therefore the returned AnalysisResult will need to go back to the db to get the attributes back out. During ingest, this can be a performance constraint.

Reply:
@jayaramcs and @raman-bt, is the plan to just pass the attributes through to the new SleuthkitCase.newAnalysisResult() method?

Reply:
@rcordovano @raman-bt I have created a new PR https://github.com/sleuthkit/sleuthkit/pull/2067 that allows for adding a blackboard artifact and its attributes in a single api call. I think it is more appropriate to use that API here.